### PR TITLE
Add permissions files for groovy-cps and workflow-cps-parent

### PR DESCRIPTION
**Pull request by dwnusbaum**
# Description

In https://github.com/jenkinsci/workflow-cps-plugin/pull/612, we are merging https://github.com/cloudbees/groovy-cps into https://github.com/jenkinsci/workflow-cps-plugin to simplify maintenance and testing. `groovy-cps` is technically set up to be released to Maven Central, but I am not even sure whether there is anyone who still has permissions set up to perform a release. The previous two releases have been released to https://repo.jenkins-ci.org/ as part of Jenkins security advisories, which bypass the permissions set by this repository.

This PR copies the [existing permissions](https://github.com/jenkins-infra/repository-permissions-updater/blob/b04a444ed9d76c5f10b62757e8569f1dacd4d7ea/permissions/plugin-workflow-cps.yml) for `workflow-cps` and applies them to the moved `groovy-cps` library and the new `workflow-cps-parent` parent POM.

I am filing this as a draft for now in case we want to change `groovy-cps`'s `groupId`.

# Submitter checklist for adding or changing permissions

### Always

- [x] Add link to plugin/component Git repository in description above

### For a new permissions file only

- [x] Make sure the file is created in `permissions/` directory
- [x] `artifactId` (pom.xml) is used for `name` (permissions YAML file).
- [x] [`groupId` / `artifactId` (pom.xml) are correctly represented in `path` (permissions YAML file)](https://github.com/jenkins-infra/repository-permissions-updater/#managing-permissions)
- [x] Check that the file is named `plugin-${artifactId}.yml` for plugins

### When adding new uploaders (this includes newly created permissions files)

- [x] [Make sure to `@`mention an existing maintainer to confirm the permissions request, if applicable](https://github.com/jenkins-infra/repository-permissions-updater/#requesting-permissions)
- [x] Use the Jenkins community (LDAP) account name in the YAML file, not the GitHub account name
- [x] Make sure to `@`mention the users being added so their GitHub account names are known if they require GitHub merge access (see below).
- [x] [All newly added users have logged in to Artifactory and Jira at least once](https://github.com/jenkins-infra/repository-permissions-updater/#requesting-permissions)

### Reviewer checklist (not for requesters!)

- [ ] Check this if newly added person also needs to be given merge permission to the GitHub repo (please @ the people/person with their GitHub username in this issue as well). If needed, it can be done using an [IRC Bot command](https://jenkins.io/projects/infrastructure/ircbot/#github-repo-management)
- [ ] Check that the `$pluginId Developers` team has `Admin` permissions while granting the access.
- [ ] In the case of plugin adoption, ensure that the Jenkins Jira default assignee is either removed or changed to the new maintainer.
- [ ] If security contacts are changed (this includes add/remove), ping the security officer (currently `@Wadeck`) in this pull request. If an email contact is changed, wait for approval from the security officer.

There are [IRC Bot commands](https://jenkins.io/projects/infrastructure/ircbot/#issue-tracker-management) for it

### CODE REVIEW

File: permissions/component-groovy-cps.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+name: "groovy-cps"
+github: "jenkinsci/workflow-cps-plugin"
+paths:
+  - "com/cloudbees/groovy-cps"
```
Comment (dwnusbaum) [inline permissions/component-groovy-cps.yml:5]: TBD: See https://github.com/jenkinsci/workflow-cps-plugin/pull/612#discussion_r1010855691.

**Comment by dwnusbaum** [comment]
Hmm, we might need to add `groovy-cps-dgm-builder` as well for incrementals even though it never gets deployed: "No permissions for com/cloudbees/groovy-cps-dgm-builder/3512.v33e276c2c53e/groovy-cps-dgm-builder-3512.v33e276c2c53e-jar-with-dependencies.jar".

**Comment by jglick** [comment]
> we might need to add `groovy-cps-dgm-builder` as well for incrementals even though it never gets deployed

Should not be necessary. There are definitely other cases of multimodule repos which skip deployment of some modules.

**Review by jglick — APPROVED** [review]

**Comment by dwnusbaum** [comment]
> Should not be necessary.

Yeah, we just need to skip installation as well, see https://github.com/jenkinsci/workflow-cps-plugin/pull/612/commits/1b01482f7077021d68c23b1c266e3b669ff83182.

File: permissions/component-groovy-cps.yml (hunk 1 above)
Comment (jglick) [reply permissions/component-groovy-cps.yml:5]: IMO this is fine as is.

**Comment by dwnusbaum** [comment]
Ok, I think this is good to go. Once this is merged we will be able to get an incremental build of https://github.com/jenkinsci/workflow-cps-plugin/pull/612 which can be used to verify https://github.com/jenkinsci/plugin-compat-tester/pull/383.

**Review by timja — APPROVED** [review]

**Pull request merged**